improver_tests/psychrometric_calculations/test_PhaseChangeLevel.py

```diff
@@ -418,6 +418,140 @@ def test_null_lat_lon(self):
+            [
+                [400.0, 500.0, 500.0, 500.0, 400.0],
+                [500.0, 700.0, 750.0, 700.0, 500.0],
+                [500.0, 700.0, 850.0, 700.0, 500.0],
+                [500.0, 700.0, 750.0, 700.0, 500.0],
+                [400.0, 500.0, 500.0, 500.0, 400.0],
+            ]
+        )
+        self.max_nbhood_orog_2d = np.array(
+            [
+                [700.0, 700.0, 700.0, 700.0, 700.0],
+                [700.0, 850.0, 850.0, 850.0, 700.0],
+                [700.0, 850.0, 850.0, 850.0, 700.0],
+                [700.0, 850.0, 850.0, 850.0, 700.0],
+                [700.0, 700.0, 700.0, 700.0, 700.0],
+            ]
+        )
+        self.expected_result_2d = np.array(
+            [
+                [1000.0, 1000.0, 950.0, 800.0, 800.0],
+                [1000.0, 1000.0, 950.0, 900.0, 800.0],
+                [1000.0, 1000.0, 950.0, 950.0, 800.0],
+                [1000.0, 1000.0, 950.0, 900.0, 800.0],
+                [1000.0, 1000.0, 950.0, 800.0, 800.0],
+            ]
+        )
+
+        # A 'nan crater' with low orography circled by high orography.
+        self.phase_change_data_2d_crater = np.full((9, 9), 1000.0)
+        self.phase_change_data_2d_crater[2:7, 2:7] = np.nan
+        self.orography_2d_crater = np.full((9, 9), 900.0)
+        self.orography_2d_crater[2:7, 2:7] = 600.0
+        self.max_nbhood_orog_2d_crater = np.full((9, 9), 900.0)
+        self.max_nbhood_orog_2d_crater[3:6, 3:6] = 600.0
+        self.expected_result_2d_crater = np.full((9, 9), 1000.0)
+
+    def test_interpolate_edge_case_1d(self):
+        """Test that we still fill in missing areas under a 1d peaked edge case."""
+        plugin = PhaseChangeLevel(phase_change="snow-sleet", grid_point_radius=1)
+        result = plugin._horizontally_interpolate_phase(
+            self.phase_change_data_1d, self.orography_1d, self.max_nbhood_orog_1d
+        )
+        self.assertArrayAlmostEqual(result, self.expected_result_1d)
+
+    def test_interpolate_edge_case_2d(self):
+        """Test that we still fill in missing areas under a peaked edge case."""
+        plugin = PhaseChangeLevel(phase_change="snow-sleet", grid_point_radius=1)
+        result = plugin._horizontally_interpolate_phase(
+            self.phase_change_data_2d, self.orography_2d, self.max_nbhood_orog_2d
+        )
+        self.assertArrayAlmostEqual(result, self.expected_result_2d)
+
+    def test_interpolate_edge_case_2d_grid_point_radius_2(self):
+        """Test filling in missing areas under a radius 2 peaked edge case."""
+        plugin = PhaseChangeLevel(phase_change="snow-sleet", grid_point_radius=2)
+        max_nbhood_orog = np.full((5, 5), 850.0)
+        result = plugin._horizontally_interpolate_phase(
+            self.phase_change_data_2d, self.orography_2d, max_nbhood_orog
+        )
+        expected_result = self.expected_result_2d.copy()
+        expected_result[2][3] = self.orography_2d[2][3]  # Why different from radius 1?
```

Comment:
Increasing the grid point radius causes more source data points to be excluded from the interpolation, hence a different value comes out.